daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -338,18 +339,9 @@ class CLIConf(arguments: Array[String]) extends scallop.ScallopConf(arguments) {
       case _ => Right(Unit)
     }
 
-    validateOpt(infosetType, stream) {
-      case (Some(InfosetType.EXI), Some(true)) => Left("Streaming mode is not currently supported with EXI infosets.")
-      case _ => Right(Unit)
-    }
-
-    validateOpt(infosetType, stream) {
-      case (Some(InfosetType.EXISA), Some(true)) => Left("Streaming mode is not currently supported with EXI infosets.")
-      case _ => Right(Unit)
-    }
-
-    validateOpt(infosetType, schema) {
-      case (Some(InfosetType.EXISA), None) => Left("A schema must be specified to use schema-aware compression with EXI")
+    validateOpt(infosetType, stream, schema) {
+      case (Some(InfosetType.EXI), Some(true), _) => Left("Streaming mode is not currently supported with EXI infosets.")
+      case (Some(InfosetType.EXISA), _, None) => Left("A schema must be specified to use schema-aware compression with EXI")
```

Comment:
This is missing the case where the infosetType is InfosetType.EXSI and stream is true. Same with below.